a3p-integration/proposals/z:acceptance/auction.test.js

```diff
@@ -0,0 +1,389 @@
+    },
+  );
+};
+
+const depositCollateral = async t => {
+  const [brandsRaw, retryOptions] = await Promise.all([
+    agoric.follow('-lF', ':published.agoricNames.brand', '-o', 'text'),
+    calculateRetryUntilNextStartTime(),
+  ]);
+  const brands = Object.fromEntries(
+    marshaller.fromCapData(JSON.parse(brandsRaw)),
+  );
+
+  const offerSpec = {
+    id: config.depositor.offerId,
+    invitationSpec: {
+      source: 'agoricContract',
+      instancePath: ['auctioneer'],
+      callPipe: [['makeDepositInvitation']],
+    },
+    proposal: {
+      give: {
+        Collateral: { brand: brands.ATOM, value: 100_000_000n },
+      },
+    },
+  };
+
+  const spendAction = {
+    method: 'executeOffer',
+    offer: offerSpec,
+  };
+
+  const offer = JSON.stringify(marshaller.toCapData(harden(spendAction)));
+  t.log('OFFER', offer);
+
+  executeOffer(GOV1ADDR, offer);
+  return waitUntilOfferResult(
+    GOV1ADDR,
+    config.depositor.offerId,
+    true,
+    ambientAuthority,
+    {
+      errorMessage: 'proceeds not distributed yet',
+      ...retryOptions,
+    },
+  );
+};
+
+const checkBidsOutcome = (settledBids, t, brands) => {
+  [...settledBids]
+    .map(bidResult => bidResult.status.payouts)
+    .forEach(({ Bid, Collateral }, i) => {
+      const {
+        payouts: { Bid: outcomeBidVal, Collateral: outcomeColVal },
+      } = config.bidsOutcome[i];
+      t.is(
+        AmountMath.isEqual(
+          Bid,
+          AmountMath.make(brands.IST, scale6(outcomeBidVal)),
+        ),
+        true,
+      );
+      t.is(
+        AmountMath.isGTE(
+          Collateral,
+          AmountMath.make(brands.ATOM, scale6(outcomeColVal)),
+        ),
+        true,
+      );
+    });
+};
+
+const getCapturedPrice = async bookId => {
+  const result = await agoric.follow('-lF', `:published.auction.${bookId}`);
+  return result;
+};
+
+const checkPrice = (res, expected) => {
+  if (res.startPrice === null) return false;
+  else if (res.startPrice.numerator.value === expected) return true;
+  return false;
+};
+
+/**
+ * Calculates a set of retry options based on current auction params
+ */
+const calculateRetryUntilNextStartTime = async () => {
+  const schedule = await agoric.follow('-lF', ':published.auction.schedule');
+  const nextStartTime = parseInt(schedule.nextStartTime.absValue);
+
+  /** @type {RetryOptions} */
+  const capturePriceRetryOpts = {
+    maxRetries: Math.round((nextStartTime * 1000 - Date.now()) / 10000) + 2, // wait until next schedule
+    retryIntervalMs: 10000, // 10 seconds in ms
+  };
+
+  return capturePriceRetryOpts;
+};
+
+test.before(async t => {
+  /** @type {RetryOptions} */
+  const pushPriceRetryOpts = {
+    maxRetries: 5, // arbitrary
+    retryIntervalMs: 5000, // in ms
+  };
+
+  /** @type {RetryOptions} */
+  const bankSendRetryOpts = {
+    maxRetries: 3, // arbitrary
+    retryIntervalMs: 3000, // in ms
+  };
+
+  // Get current round id
```

Comment:
Oh, good idea. 

Please add a version of this (with `ATOM-USD` as a parameter) to synthetic chain.

Reply:
Please see https://github.com/Agoric/agoric-3-proposals/pull/183